### PR TITLE
New swagger for 2019-08-01 API

### PR DESCRIPTION
**Pull request by leonardbf**
For R6 release of RP with new API 2019-08-01
First commit contains previous 2019-07-01 as 2019-08-01.
Amendments to 08 baseline version
 - corrections to dates
 - R6 specific alterations (properties for volume replication)
 - Improve examples
 - Address noun-verb comment from previous PR https://github.com/Azure/azure-rest-api-specs/pull/7020
 - includes some minor textual corrections and amendments to older APIs.
 - update for comment on raw tag type here https://github.com/Azure/azure-rest-api-specs/issues/7447#event-2704676341

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by leonardbf** [comment]
Assistance required:
1) It seems prettify checks have been introduced/changed. It's not clear what the format error is. NPM is installed locally but prettier is not present. Running npm install as per the instructions here https://dev.azure.com/azure-sdk/public/_build/results?buildId=173321&view=logs&jobId=08b5038d-1e84-5877-88da-859df8312b69 gives "npm ERR! azure-rest-api-specs-tests@0.1.0 postinstall: `scripts/switch-to-preproduction.sh`"
2) The breaking change here https://dev.azure.com/azure-sdk/public/_build/results?buildId=173321&view=logs&j=08b5038d-1e84-5877-88da-859df8312b69 is something that intermittently occurs when we push a new API even when no change has been made. It is always on the same location but does not always occur. It would be great to get some explanation.
Thanks.

**Comment by jhendrixMSFT** [comment]
@NullMDR are you the right person to help with these issues?

**Comment by leonardbf** [comment]
> @NullMDR are you the right person to help with these issues?

I see https://github.com/Azure/azure-rest-api-specs/pull/7666#issuecomment-547754960 was produced and this resolves the problem I had running 'npm run prettier' locally. However the resulting json still fails the check when run in the pipeline. So while the first part of my first issue above is resolved the remainder of the two issues is still relevant.

**Comment by NullMDR** [comment]
@leonardbf Please rebase on the latest master and run 

npm install; npm run prettier-fix

**Comment by leonardbf** [comment]
Pull from azure upstream performed. The prettier update seems to have worked. The breaking change is still present unfortunately.

**Comment by raych1** [comment]
@qiaozha , please have a look at the breaking change failures. Looks like it complains on the parameters declaration on path item.

**Comment by qiaozha** [comment]
@leonardbf  @raych1 Currently, BreakingChange doesn't support to put "parameters" being parrellel to methods like "get", "put", "post". Because a method should be defined as an json object, but here parameters was defined as an array. which cause json parse error. For example, from line 70 at specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json. 
The reason why you don't change this part how this time the error come out was because Previously this file was forcely merged, but you've changed other part this time. The BreakingChange Tool will check the whole file and report it again.
This is an acknowledged bug for BreakingChange Tool and my suggestion is to move the "parameter" block into the methods separately like this example move "parameters" block at line 70  into "post" block at line 91. 
Otherwise. you need to merge it forcely like previously. but in this way, before this error being fixed,  each time you change this file, it will come out.
This kind of code style can be seen in several places.

**Comment by leonardbf** [comment]
> @leonardbf @raych1 Currently, BreakingChange doesn't support to put "parameters" being parrellel to methods like "get", "put", "post". Because a method should be defined as an json object, but here parameters was defined as an array. which cause json parse error. For example, from line 70 at specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json.
> The reason why you don't change this part how this time the error come out was because Previously this file was forcely merged, but you've changed other part this time. The BreakingChange Tool will check the whole file and report it again.
> This is an acknowledged bug for BreakingChange Tool and my suggestion is to move the "parameter" block into the methods separately like this example move "parameters" block at line 70 into "post" block at line 91.
> Otherwise. you need to merge it forcely like previously. but in this way, before this error being fixed, each time you change this file, it will come out.
> This kind of code style can be seen in several places.

Ok, thanks for the update. Since this has been pushed before with this present would it be acceptable to do so again? We would address this problem in our API 2019-10-01 which we expect to push at the end of November. The 2019-08-01 is needed for the RP that went to production last week.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json [hunk 1]
```diff
@@ -0,0 +1,2415 @@
...
+        },
+        "mountTargets": {
+          "title": "mountTargets",
+          "description": "List of mount targets",
+          "items": {
+            "$ref": "#/definitions/mountTargetList"
+          }
+        },
+        "volumeType": {
+          "type": "string",
```
Comment (majastrz) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json]: >string [](start = 19, length = 6)

Should this be an enum?

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json [hunk 2]
```diff
@@ -0,0 +1,2415 @@
...
+          "description": "What type of volume is this",
+          "example": "DataProtection"
+        },
+        "dataProtection": {
```
Comment (majastrz) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json]: >dataProtection [](start = 9, length = 14)

Is this specific only to DataProtection volume type? What happens with other volume types?

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json [hunk 3]
```diff
@@ -0,0 +1,2415 @@
...
+        },
+        "endpointType": {
+          "title": "endpointType",
+          "type": "string",
+          "description": "Indicates whether the local volume is the source or destination for the Volume Replication",
+          "example": "src, dst"
+        },
+        "replicationSchedule": {
+          "title": "replicationSchedule",
+          "type": "string",
```
Comment (majastrz) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json]: >string [](start = 19, length = 6)

Should this be an enum? (Using the x-ms-enum extension you can still model the enum as a string, but the generated SDKs will include a class with constants representing each possible value. It makes it more discoverable for the SDK user.)

**Review by majastrz — COMMENTED** [review]
I added a couple of questions. Can you take a look?

**Comment by qiaozha** [comment]
> @leonardbf @raych1 Currently, BreakingChange doesn't support to put "parameters" being parrellel to methods like "get", "put", "post". Because a method should be defined as an json object, but here parameters was defined as an array. which cause json parse error. For example, from line 70 at specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json.
> The reason why you don't change this part how this time the error come out was because Previously this file was forcely merged, but you've changed other part this time. The BreakingChange Tool will check the whole file and report it again.
> This is an acknowledged bug for BreakingChange Tool and my suggestion is to move the "parameter" block into the methods separately like this example move "parameters" block at line 70 into "post" block at line 91.
> Otherwise. you need to merge it forcely like previously. but in this way, before this error being fixed, each time you change this file, it will come out.
> This kind of code style can be seen in several places.
> 
> Ok, thanks for the update. Since this has been pushed before with this present would it be acceptable to do so again? We would address this problem in our API 2019-10-01 which we expect to push at the end of November. The 2019-08-01 is needed for the RP that went to production last week.

@yungezz  Catherine, this PR BreakingChange error need to be force merged. Could you pls take a look into it? Thanks

**Comment by yungezz** [comment]
> > @leonardbf @raych1 Currently, BreakingChange doesn't support to put "parameters" being parrellel to methods like "get", "put", "post". Because a method should be defined as an json object, but here parameters was defined as an array. which cause json parse error. For example, from line 70 at specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-05-01/netapp.json.
> > The reason why you don't change this part how this time the error come out was because Previously this file was forcely merged, but you've changed other part this time. The BreakingChange Tool will check the whole file and report it again.
> > This is an acknowledged bug for BreakingChange Tool and my suggestion is to move the "parameter" block into the methods separately like this example move "parameters" block at line 70 into "post" block at line 91.
> > Otherwise. you need to merge it forcely like previously. but in this way, before this error being fixed, each time you change this file, it will come out.
> > This kind of code style can be seen in several places.
> > Ok, thanks for the update. Since this has been pushed before with this present would it be acceptable to do so again? We would address this problem in our API 2019-10-01 which we expect to push at the end of November. The 2019-08-01 is needed for the RP that went to production last week.
> 
> @yungezz Catherine, this PR BreakingChange error need to be force merged. Could you pls take a look into it? Thanks

got it. once ARM review signed off, I can do merge.

**Comment by leonardbf** [comment]
> I added a couple of questions. Can you take a look?

@majastrz I responded to the questions and hope this can be signed off soon. Thanks.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json (hunk 3 above)
Comment (majastrz) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json]: Yeah, in JSON enums look like strings. I'm ok leaving it for later.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json (hunk 2 above)
Comment (majastrz) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json]: Sounds good.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json (hunk 1 above)
Comment (majastrz) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-08-01/netapp.json]: ok

**Review by majastrz — APPROVED** [review]
Looks good from ARM side.

**Comment by jhendrixMSFT** [comment]
@leonardbf just to confirm, this API version is now live?

**Comment by leonardbf** [comment]
> @leonardbf just to confirm, this API version is now live?

Yep, this went live last week.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**